crates/apollo-compiler/tests/extensions.rs

```diff
@@ -92,6 +92,25 @@ fn test_invalid_orphan_extensions_schema_def_with_duplicate_root_operation() {
     assert!(err.contains("duplicate definitions for the `query` root operation type"));
 }
 
+#[test]
+fn test_orphan_schema_extension_disables_implicit_root_types() {
+    let input = r#"
+        extend schema { query: Query }
+        type Query { viruses: [Virus] }
+        type Virus { mutations: [Mutation] }
+        type Mutation { something: String }
+    "#;
+
+    let schema = Schema::builder()
+        .adopt_orphan_extensions()
+        .parse(input, "schema.graphql")
+        .build()
+        .unwrap();
+
+    assert!(schema.schema_definition.mutation.is_none());
```

Comment:
So in this mode, if I've explicitly defined a root operation on an extension then we shouldn't implicitly provide the others, seems reasonable.

And in the case where we're in this mode, AND I've provided an extension, AND the extension's roots are empty, then we do still provide the implicit 👍